Trimble.Modus.Components/Controls/TMInput/CustomInput.xaml.cs

```diff
@@ -0,0 +1,551 @@
+    /// Gets or sets the command for left icon
+    /// </summary>
+    public static readonly BindableProperty LeftIconCommandProperty =
+        BindableProperty.Create(nameof(LeftIconCommand), typeof(ICommand), typeof(CustomInput), null);
+
+    /// <summary>
+    /// Gets or sets the command property for left icon
+    /// </summary>
+    public static readonly BindableProperty LeftIconCommandParameterProperty =
+        BindableProperty.Create(nameof(LeftIconCommandParameterProperty), typeof(object), typeof(CustomInput), null);
+
+    /// <summary>
+    /// Gets or sets the command for right icon
+    /// </summary>
+    public static readonly BindableProperty RightIconCommandProperty =
+        BindableProperty.Create(nameof(RightIconCommandProperty), typeof(ICommand), typeof(CustomInput), null);
+
+    /// <summary>
+    /// Gets or sets the command property for right icon
+    /// </summary>
+    public static readonly BindableProperty RightIconCommandParameterProperty =
+        BindableProperty.Create(nameof(RightIconCommandParameterProperty), typeof(object), typeof(CustomInput), null);
+
+    /// <summary>
+    /// Gets or sets value that indicates whether the input control is enabled or not.
+    /// </summary>
+    public static new readonly BindableProperty IsEnabledProperty =
+        BindableProperty.Create(nameof(IsEnabled), typeof(bool), typeof(CustomInput), true, propertyChanged: OnEnabledPropertyChanged);
+
+    /// <summary>
+    /// Gets or sets value that indicates whether the input control is readonly or not.
+    /// </summary>
+    public static readonly BindableProperty IsReadOnlyProperty =
+        BindableProperty.Create(nameof(IsReadOnly), typeof(bool), typeof(CustomInput), false, propertyChanged: OnReadOnlyPropertyChanged);
+
+    /// <summary>
+    /// Gets or sets the font size property of the entry
+    /// </summary>
+    public static readonly BindableProperty FontSizeProperty =
+       BindableProperty.Create(nameof(FontSize), typeof(double), typeof(CustomInput), (double)Enums.FontSize.Small);
+
+    /// <summary>
+    /// Gets or sets the font family property of the entry
+    /// </summary>
+    public static readonly BindableProperty FontFamilyProperty =
+        BindableProperty.Create(nameof(FontFamily), typeof(string), typeof(CustomInput), null, BindingMode.TwoWay, null);
+
+    #endregion
+
+    #region Public properties
+    /// <summary>
+    /// Delegate for the input validation function
+    /// </summary>
+    /// <param name="sender">Reference to the sender input control</param>
+    /// <returns>Tuple of bool and string, bool represent the result of the text validation and string represent the text to be displayed</returns>
+    public delegate Tuple<bool, string> InputValidationHandler(object sender);
+
+    ///// <summary>
+    ///// Public event handler to be invoked when text is changed
+    ///// </summary>
+    //public event EventHandler<TextChangedEventArgs> TextChanged;
```

Comment:
Is this got removed now? We need a listener and a command for `TextChanged` event.

Reply:
Added event changed